### PR TITLE
switch from open to io.open for text file read/write

### PR DESCRIPTION
**Pull request by indirectlylit**
### Summary

In https://github.com/learningequality/kolibri-installer-windows/issues/72 we noticed that characters aren't being rendered properly in Windows using Python 3

This PR attempts to find _all_ instances of `open` that were used on text files, and replace them with `io.open`, using an explicit encoding.

### Reviewer guidance

Please check that this all makes sense

### References

* https://github.com/learningequality/kolibri/commit/715c2907e97e1937f61d1f13c8cbde3273553b31
* https://github.com/learningequality/kolibri/commit/e8d9aead2554dc5792cd9565be864a9e846cfd56
* https://github.com/learningequality/kolibri/pull/3743

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: build_tools/customize_build.py [hunk 1]
```diff
@@ -26,11 +27,11 @@ def load_plugins_from_file(file_path):
         if file_path.startswith('http'):
             print("Downloading plugins manifest from {file_path}".format(file_path=file_path))
             _, path = tempfile.mkstemp(suffix=".txt", text=True)
-            with open(path, 'w') as f:
+            with io.open(path, mode='w', encoding='utf-8') as f:
                 r = requests.get(file_path)
                 f.write(r.content)
             file_path = path
-        with open(file_path, 'r') as f:
+        with io.open(file_path, mode='r', encoding='utf-8') as f:
```
Comment (mrpau-richard) [inline build_tools/customize_build.py]: @indirectlylit I think you need to use the `KWARGS_IO_READ` when reading file. `mode='w', encoding='utf-8'` will only work at the `Python 3`.
```
# Use encoded text for Python 3 (doesn't work in Python 2!)
KWARGS_IO_READ = {'mode': 'r', 'encoding': 'utf-8'}
KWARGS_IO_WRITE = {'mode': 'w', 'encoding': 'utf-8'}

# Use binary file mode for Python 2 (doesn't work in Python 3!)
if sys.version_info < (3,):
    KWARGS_IO_READ = {'mode': 'rb'}
    KWARGS_IO_WRITE = {'mode': 'wb'}
```

**Review by mrpau-richard — CHANGES_REQUESTED** [review]

File: build_tools/customize_build.py (hunk 1 above)
Comment (indirectlylit) [reply build_tools/customize_build.py]: My understanding was that by switching from `open` to `io.open` I could avoid this discrepancy?

* https://docs.python.org/3.3/howto/pyporting.html#text-files
* https://stackoverflow.com/a/10975371

**Comment by indirectlylit** [comment]
hmm looks like it's not building... checking on it

File: build_tools/customize_build.py (hunk 1 above)
Comment (mrpau-richard) [reply build_tools/customize_build.py]: Yeah, I tested it. `KWARGS_IO_READ` not needed here.

**Review by mrpau-richard — APPROVED** [review]

File: build_tools/customize_build.py (hunk 1 above)
Comment (mrpau-richard) [reply build_tools/customize_build.py]: Waiting for the build to test.

**Comment by mrpau-richard** [comment]
@indirectlylit the fix confirmed at the Windows 7. Non-english characters are now displaying properly.
screen shot 2018-06-08 at 11 21 54 am

**Comment by indirectlylit** [comment]
great!

might also help with #2541

**Pull request merged**